{{cookiecutter.project_slug}}/fabfile.py

```diff
@@ -1,144 +1,233 @@
 from io import StringIO
 
 import dj_database_url
-from fabric.api import (
-    cd,
-    env,
-    execute,
-    get,
-    local,
-    put,
-    require,
-    run,
-    settings,
-    shell_env,
-    task,
-)
-from fabric.context_managers import quiet
-from fabric.operations import prompt
-from gitric import api as gitric
-
-# This is the definition of your environments. Every item of the ENVIRONMENTS
-# dict will be made available as a fabric task and the properties you put in a
-# particular environment will be made available in the `env` variable.
+from dulwich import porcelain
+from fabric import task
+from fabric.connection import Connection
+from invoke import Exit
+from invoke.exceptions import UnexpectedExit
+
 ENVIRONMENTS = {
     "prod": {
         "root": "/var/www/{{ cookiecutter.project_slug }}/prod/",
-        "hosts": ["root@myhost"],
+        "host": "root@myhost",
         "pid": "/path/to/uwsgi/pid",
         # You can set settings that will be automatically deployed when running
         # the `bootstrap` command
-        # 'settings': {
-        #     'ALLOWED_HOSTS': 'www.myhost.com',
-        # }
+        "settings": {
+            #     'ALLOWED_HOSTS': 'www.myhost.com',
+        },
     },
     "dev": {
-        "root": "/var/www/{{ cookiecutter.project_slug }}/staging/",
-        "hosts": ["root@myhost"],
+        "root": "/var/www/{{ cookiecutter.project_slug }}/dev/",
+        "host": "root@myhost",
         "pid": "/path/to/uwsgi/pid",
         # You can set settings that will be automatically deployed when running
         # the `bootstrap` command
-        # 'settings': {
-        #     'ALLOWED_HOSTS': 'www.myhost.com',
-        # }
+        "settings": {
+            #     'ALLOWED_HOSTS': 'www.myhost.com',
+        },
     },
 }
 
-env.project_name = "{{ cookiecutter.project_slug }}"
+project_name = "{{ cookiecutter.project_slug }}"
 
 
-def ls(path):
+def remote(task_func):
     """
-    Return the list of the files in the given directory, omitting . and ...
+    Decorate task functions to check for presence of a Connection instance in their context.
+    Also pass the Connection instance in argument for convenience.
     """
-    with cd(path), quiet():
-        files = run("for i in *; do echo $i; done")
-        files_list = files.replace("\r", "").split("\n")
-
-    return files_list
 
+    @functools.wraps(task_func)
+    def call_task_with_connection(ctx, *args, **kwargs):
+        if not hasattr(ctx, "conn"):
+            raise RuntimeError("Trying to run a remote task with no environment loaded")
+        return task_func(ctx, *args, **kwargs)
 
-def git_push(commit):
-    """
-    Push the current tree to the remote server and reset the remote git
-    repository to the given commit. The commit can be any git object, be it a
-    hash, a tag or a branch.
-    """
-    gitric.git_seed(get_project_root(), commit)
-    gitric.git_reset(get_project_root(), "master")
+    return call_task_with_connection
```

Comment:
Fabric tries to imply CLI arguments from function arguments, but it does not work with decorated functions.
Adding `call_task_with_connection.__signature__ = inspect.signature(task_func)` fixes this issue.